core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Authenticator.java

```diff
@@ -78,9 +78,12 @@ protected Optional<Identity> createIdentity(String token)
         if (claims.isEmpty()) {
             return Optional.empty();
         }
-        String principal = (String) claims.get().get(principalField);
-        Identity.Builder builder = Identity.forUser(userMapping.mapUser(principal));
-        builder.withPrincipal(new BasicPrincipal(principal));
+        Optional<String> principal = Optional.ofNullable((String) claims.get().get(principalField));
```

Comment:
Not related to this change, but why not `claims.get().get(principalField, String.class)`?

Reply:
Claims is a map of`<String, Object>` here, not `ClaimsSet`.